src/runtime/particle-execution-context.ts

```diff
@@ -235,36 +235,49 @@ export class ParticleExecutionContext {
     }];
   }
 
-  private async reloadParticle(id: string) {
-    let resolve: Runnable;
-    const p = new Promise<void>(res => resolve = res);
-    this.pendingLoads.push(p);
-
-    // Get the old particle based on the given id and delete the old particle's cache
-    const oldParticle = this.particles.get(id);
-    delete oldParticle.spec.implBlobUrl;
-
-    // Create a new particle and replace the old one
-    const particle: Particle = await this.createParticleFromSpec(id, oldParticle.spec);
-
-    const handleMap = new Map();
-    const registerList: {proxy: Store, particle: Particle, handle: Handle}[] = [];
-    // Create new handles and disable the handles of the old particles
-    oldParticle.handles.forEach((oldHandle) => {
-      this.createHandle(particle, oldParticle.spec, id, oldHandle.name, oldHandle.storage, handleMap, registerList);
-      if (oldHandle instanceof HandleOld) oldHandle.disable(oldParticle);
+  private async reloadParticle(ids: string[]) {
+    // Delete old particles' caches
+    ids.forEach(id => {
+      const oldParticle = this.particles.get(id);
+      if (oldParticle.spec.implBlobUrl) delete oldParticle.spec.implBlobUrl;
+      if (oldParticle.spec.implFile.endsWith('.wasm')) {
+        this.wasmContainers[oldParticle.spec.implFile] = undefined;
+      }
```

Comment:
This will break if there is more than one particle instantiated in the container. You can detect this by looking at WasmContainer.particleMap. There's a couple of options for this:
1. When a wasm particle is selected for reloading, grab all of the other ones mapped into its container and reload them as well.
2. Kill the selected one in the existing container, spin up a new container for the same implFile and boot the particle in that (leaving the other ones running in the existing container).

#1 is a lot easier, and I would suggest doing that for now.

Reply:
Yes, I think that will work.